src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -28,9 +29,12 @@ public Manufacturer create(Manufacturer manufacturer) {
             preparedStatement.setString(2, manufacturer.getCountry());
             preparedStatement.executeUpdate();
             ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
-            setKey(manufacturer, generatedKeys);
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
         } catch (SQLException e) {
-            throw new RuntimeException("Can't operate with DB", e);
+            throw new DataProcessingException("Can't operate with DB", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't insert to DB " + manufacturer, e);
```
all messages should be informative